src/EnergyPlus/VariableSpeedCoils.cc

```diff
@@ -4695,11 +4695,10 @@ namespace VariableSpeedCoils {
                         SupTemp -= FanCoolLoad / (CpAir * rhoair * VolFlowRate);
                     }
                     MixWetBulb = Psychrometrics::PsyTwbFnTdbWPb(state, MixTemp, MixHumRat, state.dataEnvrn->StdBaroPress, RoutineName);
-                    // need to use OutTemp for air-cooled and RatedInletWaterTemp for water-cooled
-                    if (varSpeedCoil.CondenserInletNodeNum != 0) {
-                        RatedSourceTempCool = RatedInletWaterTemp;
-                    } else {
+                    if (varSpeedCoil.CondenserType == DataHeatBalance::RefrigCondenserType::Air) {
                         RatedSourceTempCool = OutTemp;
+                    } else {
+                        RatedSourceTempCool = GetVSCoilRatedSourceTemp(state, DXCoilNum, ErrorsFound);
```

Comment:
Only coils with a condenser inlet node input were using `RatedInletWaterTemp` (29.4444C) as the `RatedSourceTempCool`, `Coil:Cooling:WaterToAirEquationFitVariableSpeed` objects don't have such inputs so their `RatedInletWaterTemp` was set to `OutTemp`, the OAT at design conditions. Now, the intended value is returned by the function, and `OutTemp` is still used by air-cooled coils.

Reply:
Similarly, `Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed` don't have a condenser inlet node so `OutTemp` was used. However, the model is expecting a condenser inlet water temperature as the second independent variables for the capacity curve as a function of the temperature, not an outdoor air temperature. This will trigger diffs for files using `Coil:WaterHeating:AirToWaterHeatPump:VariableSpeed` objects.

Reply:
Good